backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java

```diff
@@ -0,0 +1,354 @@
+   */
+  private InternalFuture<Map<String, Map.Entry<BlobExpanded, String>>> validateVersioningEntity(
+      VersioningEntry versioningEntry) {
+
+    final var futureTask =
+        InternalFuture.runAsync(
+            () -> {
+              String errorMessage = null;
+              if (versioningEntry.getRepositoryId() == 0L) {
+                errorMessage = "Repository Id not found in VersioningEntry";
+              } else if (versioningEntry.getCommit().isEmpty()) {
+                errorMessage = "Commit hash not found in VersioningEntry";
+              }
+
+              if (errorMessage != null) {
+                throw new ModelDBException(errorMessage, Code.INVALID_ARGUMENT);
+              }
+            },
+            executor);
+
+    return futureTask.thenCompose(
+        unused -> {
+          Map<String, Map.Entry<BlobExpanded, String>> requestedLocationBlobWithHashMap =
+              new HashMap<>();
+          try (Session session = modelDBHibernateUtil.getSessionFactory().openSession()) {
+            RepositoryIdentification repositoryIdentification =
+                RepositoryIdentification.newBuilder()
+                    .setRepoId(versioningEntry.getRepositoryId())
+                    .build();
+            CommitEntity commitEntity =
+                commitDAO.getCommitEntity(
+                    session,
+                    versioningEntry.getCommit(),
+                    (session1) ->
+                        repositoryDAO.getRepositoryById(session, repositoryIdentification));
+            if (!versioningEntry.getKeyLocationMapMap().isEmpty()) {
+              Map<String, Map.Entry<BlobExpanded, String>> locationBlobWithHashMap =
+                  blobDAO.getCommitBlobMapWithHash(
+                      session,
+                      commitEntity.getRootSha(),
+                      new ArrayList<>(),
+                      Collections.emptyList());
+              for (Map.Entry<String, Location> locationBlobKeyMap :
+                  versioningEntry.getKeyLocationMapMap().entrySet()) {
+                String locationKey =
+                    String.join("#", locationBlobKeyMap.getValue().getLocationList());
+                if (!locationBlobWithHashMap.containsKey(locationKey)) {
+                  return InternalFuture.failedStage(
+                      new ModelDBException(
+                          "Blob Location '"
+                              + locationBlobKeyMap.getValue().getLocationList()
+                              + "' for key '"
+                              + locationBlobKeyMap.getKey()
+                              + "' not found in commit blobs",
+                          Code.INVALID_ARGUMENT));
+                }
+                requestedLocationBlobWithHashMap.put(
+                    locationKey, locationBlobWithHashMap.get(locationKey));
+              }
+            }
+          } catch (Exception ex) {
+            return InternalFuture.failedStage(ex);
+          }
```

Comment:
Because you're inside of a future, this is already handled for you.

Reply:
I have tried to remove this catch but the code from try throwing error which is not auto handled by future.